packages/plugin-core/src/windowWatcher.ts

```diff
@@ -52,11 +54,15 @@ export class WindowWatcher {
       const ts = _.toInteger(_.trim(tsMatch[0].split(":")[1], `'" `));
 
       const dt = DateTime.fromMillis(ts);
+      const tsConfig = getConfigValue(
+        CodeConfigKeys.DEFAULT_TIMESTAMP_DECORATION_FORMAT
+      ) as string;
+      const formatOption = (DateTime as any)[tsConfig];
```

Comment:
I'm not entirely sure if I should be doing this here. Would it defeat the purpose of using typescript?

Reply:
the way to get around this is to make a `DateTimeFormat` type that consists of the union of all the static members.

```ts
type DateTimeFormat = "DATETIME_FULL" | "..." | "...";
```

Reply:
Lovely.